iris_grib/tests/unit/save_rules/test_set_fixed_surfaces.py

```diff
@@ -118,10 +120,89 @@ def test_height_point(self, mock_set):
         mock_set.assert_any_call(grib, "scaleFactorOfFirstFixedSurface", 0)
         mock_set.assert_any_call(grib, "scaledValueOfFirstFixedSurface", 12345)
         mock_set.assert_any_call(grib, "typeOfSecondFixedSurface", 255)
-        mock_set.assert_any_call(grib, "scaleFactorOfSecondFixedSurface", 255)
+        mock_set.assert_any_call(grib, "scaleFactorOfSecondFixedSurface",
+                                 GRIB_MISSING_LONG)
         mock_set.assert_any_call(grib, "scaledValueOfSecondFixedSurface",
                                  GRIB_MISSING_LONG)
 
+    def test_unknown_vertical_unbounded(self):
+        cube = iris.cube.Cube([0])
+        cube.add_aux_coord(
+            iris.coords.AuxCoord([1],
+                                 attributes={'GRIB_fixed_surface_type': 151}))
+        grib = gribapi.grib_new_from_samples("GRIB2")
+        with warnings.catch_warnings(record=True) as warn:
+            set_fixed_surfaces(cube, grib)
+
+        self.assertEqual(len(warn), 1)
+        message = "vertical-axis coordinate unit may not be encoded correctly"
+        self.assertRegex(str(warn[0].message), message)
+
+        self.assertEqual(gribapi.grib_get_long(
+            grib, "typeOfFirstFixedSurface"), 151)
+        self.assertEqual(gribapi.grib_get_double(
+            grib, "scaledValueOfFirstFixedSurface"), 1)
+        self.assertEqual(gribapi.grib_get_double(
+            grib, "scaleFactorOfFirstFixedSurface"), 0)
+        self.assertEqual(gribapi.grib_get_long(
+            grib, "typeOfSecondFixedSurface"), 255)
+        self.assertEqual(gribapi.grib_get_long(
+            grib, "scaledValueOfSecondFixedSurface"), GRIB_MISSING_LONG)
+        self.assertEqual(gribapi.grib_get_long(
+            grib, "scaleFactorOfSecondFixedSurface"), GRIB_MISSING_LONG)
+
+    def test_unknown_vertical_bounded(self):
+        cube = iris.cube.Cube([0])
+        cube.add_aux_coord(
+            iris.coords.AuxCoord([450], bounds=np.array([900.0, 0.0]),
```

Comment:
An upper bound value of 0.0 feels a bit like it _could_ be a default or a coincidence, when we check for it  in the output.
Could you maybe make it a more unlikely value, just for a bit more certainty?

Reply:
:+1: